cmd/contour/ingressstatus.go

```diff
@@ -14,32 +14,35 @@
 package main
 
 import (
+	"context"
 	"net"
 	"strings"
-	"sync"
 
 	contour_api_v1 "github.com/projectcontour/contour/apis/projectcontour/v1"
 	"github.com/projectcontour/contour/internal/k8s"
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// loadBalancerStatusWriter manages the lifetime of IngressStatusUpdaters.
+// loadBalancerStatusWriter orchestrates LoadBalancer address status
+// updates for HTTPProxy and Infress objects.
 //
 // The theory of operation of the loadBalancerStatusWriter is as follows:
+//
 // 1. On startup the loadBalancerStatusWriter waits to be elected leader.
 // 2. Once elected leader, the loadBalancerStatusWriter waits to receive a
```

Comment:
it'd be helpful to also document that once elected leader, it creates a `StatusAddressUpdater` which will handle writing LBStatus for future Ingress/HTTPProxy objects.

Reply:
I added this info, though in a slightly different place.